lib/net/pool.js

```diff
@@ -3456,20 +3442,29 @@ class Pool extends EventEmitter {
    */
 
```

Comment:
The call to `addLoader` could increase the number of outbound peers, so calculating `need` before `addLoader` can cause you to overestimate the number of peers that you need and cause you to connect to an extra peer.